com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeSubdivide.compute

```diff
@@ -0,0 +1,261 @@
+// Each #kernel tells which function to compile; you can have many kernels
+#pragma kernel Clear
+#pragma kernel ClearBuffer
+#pragma kernel FillUVMap
+#pragma kernel JumpFlooding
+#pragma kernel FinalPass
+#pragma kernel VoxelizeProbeVolumeData 
+#pragma kernel Subdivide
+
+#pragma enable_d3d11_debug_symbols
+
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Color.hlsl"
+#include "Packages/com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbePlacement.cs.hlsl"
+
+Texture3D _Input;
+Texture3D _ProbeVolumeData;
+RWTexture3D<float4> _Output;
+RWTexture3D<float4> _FinalOutput;
+
+AppendStructuredBuffer<float3> _Bricks;
+RWStructuredBuffer<float3> _BricksToClear;
+
+StructuredBuffer<GPUProbeVolumeOBB> _ProbeVolumes;
+
+float3 _Size;
+float _Offset;
+float3 _MaxBrickSize;
+float3 _VolumeSizeInBricks;
+float3 _VolumeOffsetInBricks;
+float3 _VolumeWorldOffset;
+float _BrickSize;
+float3 _SDFSize;
+float _SubdivisionLevel;
+float _MaxSubdivisionLevel;
+float _ProbeVolumeCount;
+
+[numthreads(8,8,1)]
+void Clear(uint3 id : SV_DispatchThreadID)
+{
+    _Output[id] = 0;
+}
+
+[numthreads(64,1,1)]
+void ClearBuffer(uint3 id : SV_DispatchThreadID)
+{
+    _BricksToClear[id.x] = 0;
+}
+
+[numthreads(8,8,1)]
+void FillUVMap(uint3 id : SV_DispatchThreadID)
+{
+    float4 input = _Input[id];
+
+    if (input.a > 0.5)
+    {
+        _Output[id] = float4(id / _Size.x, 1);
+    }
+    else
+    {
+        _Output[id] = 0; // mark UV as invalid with w = 0
+    }
+}
+
+static float3 offset3D[27] =
+{
+    float3(-1, -1, -1),
+    float3(-1,  0, -1),
+    float3(-1,  1, -1),
+    float3( 0, -1, -1),
+    float3( 0,  0, -1),
+    float3( 0,  1, -1),
+    float3( 1, -1, -1),
+    float3( 1,  0, -1),
+    float3( 1,  1, -1),
+
+    float3(-1, -1,  0),
+    float3(-1,  0,  0),
+    float3(-1,  1,  0),
+    float3( 0, -1,  0),
+    float3( 0,  0,  0),
+    float3( 0,  1,  0),
+    float3( 1, -1,  0),
+    float3( 1,  0,  0),
+    float3( 1,  1,  0),
+
+    float3(-1, -1,  1),
+    float3(-1,  0,  1),
+    float3(-1,  1,  1),
+    float3( 0, -1,  1),
+    float3( 0,  0,  1),
+    float3( 0,  1,  1),
+    float3( 1, -1,  1),
+    float3( 1,  0,  1),
+    float3( 1,  1,  1),
+};
+
+float Distance(float3 pos)
+{
+    return length(pos);
+}
+
+[numthreads(8,8,1)]
+void JumpFlooding(uint3 id : SV_DispatchThreadID)
+{
+    float4 nearest = _Input[id];
+
+    if (nearest.w < 0.5)
+        nearest = float4(-10, -10, -10, 0);
+
+    int j = 0;
+    for (int i = 0; i < 27; i++)
+    {
+        float3 o = id + offset3D[i] * _Offset;
+
+        o %= _Size.x;
```

Comment:
If this is unsigned, casting with (uint) the size before the modulo will help perf.